pkg/registries/registries_test.go

```diff
@@ -914,31 +909,332 @@ func TestEditRegistriesConfig(t *testing.T) {
 			itmss := []*apicfgv1.ImageTagMirrorSet{}
 			if tt.idmsNotEmpty {
 				idmss = append(idmss, &apicfgv1.ImageDigestMirrorSet{
-
 					Spec: apicfgv1.ImageDigestMirrorSetSpec{
 						ImageDigestMirrors: []apicfgv1.ImageDigestMirrors{
 							{Source: "registry-b.com", Mirrors: []apicfgv1.ImageMirror{"mirror-digest-1.registry-b.com"}},
 						},
-					}})
+					},
+				})
 			}
 			if tt.itmsNotEmpty {
 				itmss = append(itmss, &apicfgv1.ImageTagMirrorSet{
 					Spec: apicfgv1.ImageTagMirrorSetSpec{
 						ImageTagMirrors: []apicfgv1.ImageTagMirrors{
 							{Source: "registry-c.com", Mirrors: []apicfgv1.ImageMirror{"mirror-tag-1.registry-c.com"}},
 						},
-					}})
+					},
+				})
 			}
 			if tt.icspNotEmpty {
 				icsps = append(icsps, &apioperatorsv1alpha1.ImageContentSourcePolicy{
 					Spec: apioperatorsv1alpha1.ImageContentSourcePolicySpec{
 						RepositoryDigestMirrors: []apioperatorsv1alpha1.RepositoryDigestMirrors{
 							{Source: "registry-a.com", Mirrors: []string{"mirror-icsp-1.registry-a.com"}},
 						},
-					}})
+					},
+				})
 			}
 			err = EditRegistriesConfig(&config, nil, nil, icsps, idmss, itmss)
-			assert.NotNil(t, err)
+			assert.Nil(t, err)
+		})
+	}
+}
+
+func TestEditRegistriesConfigWithBothICSPAndIDMS(t *testing.T) {
```

Comment:
Couldn’t all the new tests below be added to the existing `TestEditRegistriesConfig` test data? (I didn’t carefully compare the test executor bodies, maybe there is a significant difference.)

Alternatively, consider if some parts of the tests could share some helper functions.

Reply:
`TestEditRegistriesConfig` has a test body where the framework allows for different permutations. I preserved `TestEditRegistriesConfig` to allow for more permutations to be added.

Reply:
Fixed. Thanks!